addons/website/static/src/js/editor/widget_link.js

```diff
@@ -103,5 +104,22 @@ weWidgets.LinkTools.include({
         this._super.apply(this, arguments);
         this._adaptPageAnchor();
     },
+    /**
+     * @override
+     * @param {Event} ev
+     */
+    _onPickSelectOption(ev) {
+        // For the anchor selector, we do not adapt the preview in this event
+        // but rather in "_onAnchorChange" because we have to wait for the
+        // URL input to be updated with the anchor before updating the link in
+        // the DOM.
+        if (ev.target.parentElement.matches('[name="link_anchor"]')) {
+            // TODO: In master, we should remove the '_onAnchorChange' event
+            // and update the input URL with the anchor here.
+            this.adaptPreview = false;
```

Comment:
Can't you call _onAnchorChange here directly then? :thinking: I think it would be more stable as you would not have to add `this._adaptPreview();` in `_onAnchorChange` (and the diff would be smaller, no need of `this.adaptPreview`.

Reply:
> Can't you call _onAnchorChange here directly then? thinking I think it would be more stable as you would not have to add `this._adaptPreview();` in `_onAnchorChange` (and the diff would be smaller, no need of `this.adaptPreview`.

Indeed, I had thought about doing that but I wasn't sure if it was ok to remove the (click) event listener from the widget in stable.

Reply:
The event handler will still be called on click, so the behavior would be the same. Having one less defined event handler is indeed not *entirely* stable but more than add adding a call to `_adaptPreview` in `_onAnchorChange` like you did originally :wink: Both cases are ok though (reward >>> risk) but best choosing the alternative which has the less code (and more or less the same as the master fix will be).